src/meta/processors/zone/AddHostsProcessor.cpp

```diff
@@ -0,0 +1,70 @@
+/* Copyright (c) 2021 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#include "meta/processors/zone/AddHostsProcessor.h"
+
+#include "version/Version.h"
+
+namespace nebula {
+namespace meta {
+
+void AddHostsProcessor::process(const cpp2::AddHostsReq& req) {
+  folly::SharedMutex::ReadHolder zHolder(LockUtils::zoneLock());
+  folly::SharedMutex::ReadHolder mHolder(LockUtils::machineLock());
+  auto hosts = req.get_hosts();
+  // Confirm that there are no duplicates in the parameters.
+  if (std::unique(hosts.begin(), hosts.end()) != hosts.end()) {
+    LOG(ERROR) << "Hosts have duplicated element";
+    handleErrorCode(nebula::cpp2::ErrorCode::E_INVALID_PARM);
+    onFinished();
+    return;
+  }
+
+  // Confirm that the parameter is not empty.
+  if (hosts.empty()) {
+    LOG(ERROR) << "Hosts is empty";
+    handleErrorCode(nebula::cpp2::ErrorCode::E_INVALID_PARM);
+    onFinished();
+    return;
+  }
+
+  std::vector<kvstore::KV> data;
+  nebula::cpp2::ErrorCode code;
+  for (auto& host : hosts) {
+    // Ensure that the node is not registered.
+    auto machineKey = MetaKeyUtils::machineKey(host.host, host.port);
+    if (machineExist(machineKey) == nebula::cpp2::ErrorCode::SUCCEEDED) {
+      LOG(ERROR) << "The host " << host << " have existed!";
+      code = nebula::cpp2::ErrorCode::E_EXISTED;
+      break;
+    }
+
+    // Automatic generation zone
+    auto zoneName = folly::stringPrintf("default_zone_%s_%d", host.host.c_str(), host.port);
```

Comment:
I got this error
(root@nebula) [(none)]> rename zone default_zone_127.0.0.1_40180 to z2
[ERROR (-1004)]: SyntaxError: syntax error near `.0'

I think the zone_name should not contain "." .